File: src/main/scala/s/template_method/CharDisplay.scala
/* (C) 2022 */
package s.template_method

class CharDisplay(val ch: Char) extends AbstractDisplay {

  override def open(): Unit = {
    scala.Predef.print("<<")
  }

  override def print(): Unit = {
    scala.Predef.print(ch)
  }

  override def close(): Unit = {
    scala.Predef.println(">>")
  }

}

File: src/main/java/j/template_method/inheritance/CharDisplay.java
package j.template_method.inheritance;

public final class CharDisplay extends Display {
  private final char ch;

  CharDisplay(char ch) {
    this.ch = ch;
  }

  @Override
  public void open() {
    System.out.print("<<");
  }

  @Override
  public void print() {
    System.out.print(ch);
  }

  @Override
  public void close() {
    System.out.print(ch);
  }
}
